Library/ViewModels/PledgeViewModel.swift

```diff
@@ -449,87 +446,115 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
+    let createOrUpdateBackingEventValuesNoSCA = createOrUpdateBackingEventValues
+      .filter(requiresSCA >>> isFalse)
 
-    self.goToThanks = Signal.merge(applePayTransactionCompleted, createBackingTransactionSuccess)
+    let createOrUpdateBackingDidCompleteNoSCA = submitButtonTapped
+      .takeWhen(createOrUpdateBackingEventValuesNoSCA)
+      .filter(isTrue)
+      .ignoreValues()
 
-    // MARK: - Success/Failure Update
+    let createOrUpdateBackingEventValuesRequiresSCA = createOrUpdateBackingEventValues
+      .filter(requiresSCA)
 
-    let updatePaymentAuthorizationDidFinishSignal = willUpdateApplePayBacking
-      .takeWhen(self.paymentAuthorizationDidFinishSignal)
+    self.beginSCAFlowWithClientSecret = createOrUpdateBackingEventValuesRequiresSCA.map { $0.clientSecret }
+      .skipNil()
 
-    let updateApplePayBackingCompleted = Signal.combineLatest(
-      updateBackingEvent
-        .filter { $0.isTerminating }
-        .ignoreValues(),
-      updatePaymentAuthorizationDidFinishSignal
-    )
+    let didCompleteApplePayBacking = createOrUpdateBackingEventValues
+      .takeWhen(createOrUpdateApplePayBackingCompleted)
 
-    let updateApplePayBackingSuccess = updateBackingEvent.values()
-      .takeWhen(updateApplePayBackingCompleted)
+    let creatingContext = context.filter { $0.isCreating }
 
-    let updateBackingDidCompleteApplePay = updateApplePayBackingSuccess
-      .ignoreValues()
+    #warning("To be removed once new CreateBacking is implemented to handle Apple Pay")
+    let deprecatedCreateApplePayBackingCompleted = Signal.combineLatest(
+      deprecatedCreateApplePayBackingEvent.filter { $0.isTerminating }.ignoreValues(),
+      paymentAuthorizationDidFinish
+    )
 
-    let submitButtonTapped = Signal.merge(
-      self.submitButtonTappedSignal.mapConst(true),
-      willUpdateApplePayBacking.mapConst(false)
+    let deprecatedCreateApplePayBackingCompletedError = deprecatedCreateApplePayBackingCompleted
+      .withLatest(from: deprecatedCreateApplePayBackingEvent.errors())
+      .map(second)
+
+    let createBackingCompletionEvents = Signal.merge(
+      deprecatedCreateApplePayBackingEvent.values()
+        .combineLatest(with: deprecatedCreateApplePayBackingCompleted)
+        .ignoreValues(),
+      didCompleteApplePayBacking.combineLatest(with: willCreateApplePayBacking).ignoreValues(),
+      createOrUpdateBackingDidCompleteNoSCA.combineLatest(with: creatingContext).ignoreValues(),
+      scaFlowCompletedWithSuccess.combineLatest(with: creatingContext).ignoreValues()
     )
 
-    let updateBackingDidComplete = submitButtonTapped
-      .takeWhen(updateBackingEvent.values())
-      .filter(isTrue)
-      .ignoreValues()
+    self.goToThanks = project.takeWhen(createBackingCompletionEvents)
 
-    let updateBackingEventErrors = updateBackingEvent.errors()
+    let createUpdateBackingEventErrors = Signal.merge(
+      deprecatedCreateApplePayBackingEvent.errors().map { $0 as Error },
```

Comment:
I don't _think_ you should have to do this casting to `Error` 🤔

Reply:
I think that's because `self.showErrorBannerWithMessage` merges different types of errors (`GraphError` and `Error`). What if here we simplify to:
```
let createUpdateBackingEventErrors = Signal.merge(
      deprecatedCreateApplePayBackingEvent.errors(),
      createBackingEvent.errors(),
      updateBackingEvent.errors()
    ).map { $0 as Error }
```

Reply:
Moved the mapping to after the merge. I think I was moving things around and did this to satisfy the compiler at some point.

Once we've removed `deprecatedCreateApplePayBackingCompletedError` this should be a little simpler.

Reply:
Moved this around to better support Graph vs SCA errors 👍